gax/src/main/java/com/google/api/gax/rpc/BidiStreamingCallable.java

```diff
@@ -44,22 +44,196 @@
 
   protected BidiStreamingCallable() {}
 
+  /**
+   * The "base" method from which other forms of {@code call}s are derived. Most users will not need
+   * to call this method directly.
+   *
+   * <p>However, it is {@code public}, since library authors might want to call this method in
+   * adaptor classes.
+   */
+  public abstract ClientStream<RequestT> internalCall(
+      ResponseObserver<ResponseT> responseObserver,
+      ClientStreamReadyObserver<RequestT> onReady,
+      ApiCallContext context);
+
+  /**
+   * Listens to server responses and send requests when the network is free. Example usage:
+   *
+   * <pre>{@code
+   * final Iterator<Integer> sourceDataIterator = intCollection.iterator();
+   * BidiStreamObserver<Integer, String> bidiStreamObserver = new BidiStreamObserver<Integer, String>() {
+   *   public void onStart(StreamController controller) {
+   *     // no-op
+   *   }
+   *
+   *   public void onResponse(String response) {
+   *     System.out.println(response);
+   *   }
+   *
+   *   public void onComplete() {
+   *     System.out.println("done!");
+   *   }
+   *
+   *   public void onError(Throwable t) {
+   *     System.out.println("error: " + t);
+   *   }
+   *
+   *   public void onReady(ClientStream<Integer> stream) {
+   *     while (sourceDataIterator.hasNext()) {
+   *       if (stream.isReady()) {
+   *         stream.send(sourceDataIterator.next());
+   *       } else {
+   *         // It's OK we haven't consumed the whole iterator;
+   *         // onReady will be called again when the network becomes free.
+   *         return;
+   *       }
+   *     }
+   *     // We ran out of things to send.
+   *     stream.close();
+   *   }
+   * };
+   *
+   * bidiStreamingCallable.call(bidiStreamObserver);
+   * }</pre>
+   */
+  public void call(final BidiStreamObserver<RequestT, ResponseT> bidiObserver) {
```

Comment:
One potential issue with this:
```java
BidiStreamObserver<A,B> observer = getBidiStreamObserver();
call(observer);
```
and
```java
ResponseObserver<A,B> observer = getBidiStreamObserver();
call(observer);
```
will both compile but mean different things.

@garrettjonesgoogle Are we OK with this?

Reply:
Hmm I didn't think of that - that could be a problem given that BidiStreamObserver inherits from ResponseObserver, which means we would behavior could change purely based on the observed type as opposed to the runtime type.